src/v/kafka/server/group_manager.cc

```diff
@@ -219,7 +219,7 @@ ss::future<> group_manager::handle_partition_leader_change(
     p->loading = true;
     if (leader_id != _self.id()) {
         // TODO: we are not yet handling group / partition deletion
-        return ss::make_ready_future<>();
+        klog.trace("ignorning group/partition deletion on node {}", _self.id());
```

Comment:
> Several customers have reported time out exceptions when listing groups
when they set a default number of replications in their .yaml config
file. The culprit was a return with void future during the partition
leadership change process.

@NyaliaLui we need to add more context to ticket and/or the commit message. it's important for being able to go back and understand the change in the future (e.g. via git-blame), but also for reviewing. for example, I haven't looked at this code in many months, so I really don't know what's going on. while it's all fresh in your mind, explaining things in detail makes the review more efficient :)

we need a root cause for the issue. _when they set a default number of replications in their .yaml config_ is an important clue you used to solve the problem, but it doesn't seem like it is the root cause. why does changing that lead to some behavior? what is that problematic behavior? and finally, why does your change fix the problem?